USWDS - Dependencies: POAM Dec '23

mejiaj opened the pull request:
# Summary

Addressing two POAM issues, dependency updates, removing unused deps. Closes #5712.

**Before**: 51 vulnerabilities (24 moderate, 27 high).
**After:** 44 vulnerabilities (17 moderate, 27 high).

## Breaking change

This is not a breaking change.

## Related issue

Part of POAM process.

## Related pull requests

N/A

## Preview link

Preview link:
[StorybookJS preview → ](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/jm-poam-dec-23/?path=/story/design-tokens-fonts--fonts)

## Problem statement

Addressing two ongoing dependency issues with no available updates.

## Solution

1. Updated patch and minor dependencies.
2. Removed unused deps.
3. Added dependency that was used, but not in `package.json`.


## Major changes

**Removed `uswds-core`'s package.json file.** This experimental work unintentionally got published. Removing it in favor of open PR [USWDS - Packages: Prevent nested node_modules from being included in publish #5245](https://github.com/uswds/uswds/pull/5245) because its causing conflicts with project dependencies and sass unit testing (sass-embedded isn't a dependency of uswds-core).

## Testing and review

1. Run `npm install`.
2. Run `npm start`.
3. Update a SASS file and see it update in StorybookJS.
4. Running `npx gulp sassTests` or  `npm run test` should **not** fail.
5. Run gulp tasks (like build) and ensure there aren't errors and things build correctly.


## Dependency updates

| Dependency               | Old      | New      |
| :----------------------- | :------- | :------- |
| @babel/core              | 7.23.2   | 7.23.6   |
| @babel/preset-env        | 7.23.2   | 7.23.6   |
| @types/node              | 20.8.9   | 20.10.4  |
| eslint                   | 8.52.0   | 8.55.0   |
| eslint-config-prettier   | 9.0.0    | 9.1.0    |
| handlebars-helpers  [^1] | 0.10.0   | -        |
| html-webpack-plugin      | 5.5.3    | 5.5.4    |
| postcss                  | 8.4.31   | 8.4.32   |
| postcss-import      [^2] | -        | 15.1.0   |
| postcss-preset-env       | 9.2.0    | 9.3.0    |
| postcss-sass-loader [^3] | 1.1.0    | -        |
| sass [^4] |  - |  1.69.5 |
| snyk                     | 1.1237.0 | 1.1262.0 |
| svgo                     | 3.0.2    | 3.1.0    |
| typescript               | 5.2.2    | 5.3.3    |


[^1]: Previously used for token JSON to SASS conversion.
[^2]: Was already being used, but not included directly in package.json.
[^3]: Dependency that wasn't called directly. Hasn't been updated in 5 years now.
[^4]: Added as a dependency for `sass-true`.

Also created #5674 to address an outstanding nanoid issue.















<!--
Before opening this PR, make sure you’ve done whichever of these applies to you:
- [ ] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [ ] Run `npm run prettier:sass` to format any Sass updates.
- [ ] Run `npm test` and confirm that all tests pass.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
-->

mahoneycm commented on package.json:
```diff
@@ -150,8 +150,7 @@
     "gulp-sourcemaps": "3.0.0",
     "gulp-svgstore": "9.0.0",
     "gulp-uglify": "3.0.2",
-    "handlebars-helpers": "0.10.0",
```
Just flagging there's an [existing PR](https://github.com/uswds/uswds/pull/5599) to remove handlebars-helpers with some additional information. We can go ahead and close that PR in favor of this one 👍

mahoneycm reviewed (approved):
Lgtm!

- [x] Vulnerabilities reduced.
- [x] No build errors.
- [x] No other gulp task errors.
- [x] Style changers render in storybook correctly.
- [x] `npm audit fix` doesn't change any dependencies.

amyleadem reviewed (changes requested):
@mejiaj  I am getting an error here with the `gulp buildSass` command (as part of `npm run test`) that I am not seeing in `develop`. Are you seeing the same thing? 

Here is the error:
```
Error: Cannot find module 'sass'
Require stack:
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/sass-true/lib/main.js
- /Users/amy.leadem/Dev/github/workspace/uswds/packages/uswds-core/src/test/sass.spec.js
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/esm-utils.js
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/cli/one-and-dones.js
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/cli/options.js
- /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/bin/mocha
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.runSass (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/sass-true/lib/main.js:50:12)
    at Object.<anonymous> (/Users/amy.leadem/Dev/github/workspace/uswds/packages/uswds-core/src/test/sass.spec.js:6:10)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at exports.requireOrImport (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/esm-utils.js:42:12)
    at exports.loadFilesAsync (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/esm-utils.js:55:34)
    at Mocha.loadFilesAsync (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js:473:19)
    at singleRun (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run-helpers.js:125:15)
    at exports.runMocha (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run-helpers.js:190:10)
    at exports.handler (/Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run.js:362:11)
    at /Users/amy.leadem/Dev/github/workspace/uswds/node_modules/gulp-mocha/node_modules/yargs/build/index.cjs:443:71
[16:13:42] 'sassTests' errored after 246 ms
[16:13:42] Error in plugin "gulp-mocha"
Message:
    There were test failures
Details:
    domainEmitter: [object Object]
    domainThrown: false

[16:13:42] 'test' errored after 7.37 s
```

mejiaj commented:
### re: gulp sassTests error

@amyleadem good find, I was able to reproduce. I've added two additional changes and updated the PR description.

1. Remove unreleased work related to `uswds-core` package [^1] [d09b1ab](https://github.com/uswds/uswds/pull/5673/commits/d09b1abf4511dde549365d377e8bf2087d0fed24).
2. Install `sass` dependency for `sass-true` [^2] [b478b7f](https://github.com/uswds/uswds/pull/5673/commits/b478b7f21f8ac9b50b8e9233d39bffbcc62b93d0).

[^1]: Once merged we can close out #5245.
[^2]: Not ideal, but I couldn't find any guidance in `sass-true` docs, issues, or pr's.

amyleadem reviewed (approved):
Thanks for fixing and thanks for the footnotes. 

- [x] Confirmed vulnerability changes listed in the PR description are current (44 vulnerabilities (17 moderate, 27 high))
- [x] Confirm dependency changes in the PR description match code changes
- [x] Run fresh `npm install` without error
- [x] Run `npm start`, `npm test`, `gulp`, and `npm run build:html` without error
- [x] Confirm Sass changes compile as expected
